aws/src/test/java/org/apache/iceberg/aws/TestRESTSigV4AuthManager.java

```diff
@@ -132,16 +131,16 @@ void catalogSession() {
 
   @Test
   void contextualSession() {
-    AuthManager delegate = Mockito.mock(AuthManager.class);
-    when(delegate.catalogSession(any(), any()))
-        .thenReturn(Mockito.mock(OAuth2Util.AuthSession.class));
-    when(delegate.contextualSession(any(), any()))
-        .thenReturn(Mockito.mock(OAuth2Util.AuthSession.class));
+    AuthManager delegate =
```

Comment:
The tests failed to catch the issue because the delegate was a mock. I replaced with an actual manager, and now the tests are properly exercising the delegate, including the session cache.